plugins/uefi-capsule/make-images.py

```diff
@@ -141,8 +141,12 @@ def main(args) -> int:
                 items = Pango.itemize(pctx, label_translated, 0, length, attrs, None)
                 if not items:
                     continue
-                gs = Pango.GlyphString()
-                Pango.shape(label_translated, length, items[0].analysis, gs)
+                try:
+                    # urgh, https://gitlab.gnome.org/GNOME/pango/-/merge_requests/829
+                    gs = Pango.shape(label_translated, length, items[0].analysis)
```

Comment:
what about using `inspect`?  Something like described here:
https://stackoverflow.com/questions/3534852/is-there-a-way-to-check-a-functions-signature-in-python

Reply:
I thought about this, but for GObject introspection you only get `OrderedDict({'args': <Parameter "*args">, 'kwargs': <Parameter "**kwargs">})` for the `parameters`  --i.e. `inspect.signature(Pango.shape)` gives `(*args, **kwargs)` which is super unhelpful.

Reply:
Proper support for `inspect` in PyGObject is a recent addition: https://gitlab.gnome.org/GNOME/pygobject/-/merge_requests/341

Reply:
Okay, that's good for the future. `python3-gobject 3.50.0` is the newest I can get in Fedora 42 and 43 at the moment.